app/src/main/java/eu/kanade/tachiyomi/util/StorageUtil.kt

```diff
@@ -0,0 +1,38 @@
+package eu.kanade.tachiyomi.util
+
+import com.hippo.unifile.UniFile
+
+/**
+ * Converts a long to a readable file size.
+ */
+fun Long.toSize(): String {
+    val kb = 1024
+    val mb = kb * 1024
+    val gb = mb * 1024
+    return when {
+        this >= gb -> "%.2f GB".format(this.toFloat() / gb)
+        this >= mb -> "%.2f MB".format(this.toFloat() / mb)
+        this >= kb -> "%.2f KB".format(this.toFloat() / kb)
```

Comment:
Technically this would be GiB, MiB and KiB.
If you want to use GB, MB and KB, then divide by powers of 1000, not 1024.